superset-frontend/plugins/plugin-chart-echarts/src/utils/series.ts

```diff
@@ -642,3 +646,22 @@ export function getTimeCompareStackId(
     }) || defaultId
   );
 }
+
+const TOOLTIP_SERIES_KEY = 'seriesId';
+export function extractTooltipKeys(
+  forecastValue: any[],
+  yIndex: number,
+  richTooltip?: boolean,
+  tooltipSortByMetric?: boolean,
+): string[] {
+  if (richTooltip && tooltipSortByMetric) {
+    return forecastValue
+      .slice()
```

Comment:
Slicing to not mutate the input value (this would have caused issues in the new test)